lib/krb5/principal.c

```diff
@@ -211,6 +211,10 @@ krb5_parse_name_flags(krb5_context context,
     int enterprise = flags & KRB5_PRINCIPAL_PARSE_ENTERPRISE;
     int ignore_realm = flags & KRB5_PRINCIPAL_PARSE_IGNORE_REALM;
     int no_def_realm = flags & KRB5_PRINCIPAL_PARSE_NO_DEF_REALM;
+    int srv_and_inst = 0;
+
+    if (!strncasecmp(name, KRB5_TGS_NAME, 6))
```

Comment:
A couple of things.  First, instead of "6" the constant KRB5_TGS_NAME_SIZE should be used if it were appropriate to be using strncasecmp() but it isn't safe to do so.  What this statement compares is up to 6 characters of name to "krbtgt".  But it will succeed with if the first component of name is "krbtgt" or if it is "krbtgtxxxxxxxxxx".  

What is necessary is to ensure that the first component is a match for both string and length.

A second question that comes to mind is whether or not this is required to be a case insensitive comparison.   Although AD is not case sensitive, Kerberos is case sensitive.  I will ask therefore for some evidence that name can ever be a case variant of "krbtgt".